dnsmasq.conf

```diff
@@ -4,6 +4,11 @@
 log-queries
 #dont use hosts nameservers
 no-resolv
+# DNSSEC setup
+dnssec
+trust-anchor=.,19036,8,2,49AAC11D7B6F6446702E54A1607371607A1A41855200FD2CE1CDDE32F24E8FB5
+trust-anchor=.,20326,8,2,E06D44B80B8F1D39A95C0B0D7C65D08458E880409BBC683457104237C7F8EC8D
+dnssec-check-unsigned
```

Comment:
You should provide context about such configuration values, they should not be trusted otherwise.

The trust-anchors seem legitimate([Gentoo wiki offers the exact same snippet](https://wiki.gentoo.org/wiki/Dnsmasq/en#DNSSEC)), but rather than adding them inlined here, you should be doing it like this:

```
conf-file=/usr/share/dnsmasq/trust-anchors.conf
dnssec
```

`dnssec-check-unsigned` also looks useful.

> Now some replies are not DNSSEC signed but may still be may be legitimate, because the domain is unsigned, or may be forgeries. This setting will have dnsmasq do some various checks to see if it’s still looks valid. Now this will take a few more dns lookups, but it’s good for security. - [Source](https://www.supertechcrew.com/dnsmasq-caching-dnssec/)

> As a default, dnsmasq does not check that unsigned DNS replies are legitimate: they are assumed to be valid and passed on (without the “authentic data” bit set, of course). This does not protect against an attacker forging unsigned replies for signed DNS zones, but it is fast. If this flag is set, dnsmasq will check the zones of unsigned replies, to ensure that unsigned replies are allowed in those zones. The cost of this is more upstream queries and slower performance. - [Source](https://blog.josefsson.org/tag/dnssec/)